### PR TITLE
Update to 8.0.0 (GA)

### PR DESCRIPTION
**Pull request by tianon**
Happy :turkey: day from the PHP project, I guess? :shrug:

### CODE REVIEW

**Review by WyriHaximus — APPROVED** [review]
Happy 🐘 day

**Review by andypost — APPROVED** [review]
Looks ready to go!

**Comment by Yurunsoft** [comment]
This is very much needed
Go! Go! Go!

**Comment by dpfaffenbauer** [comment]
🎉

**Comment by MaximeMazet** [comment]
Please merge, please :'(

**Comment by eldadfux** [comment]
🎉🎈🚀

File: generate-stackbrew-library.sh
Comment (mvorisek) [inline generate-stackbrew-library.sh:7]: PHP 8.1 has no release candicated. Would it be possible to release devel (php-src master) nighty?

**Comment by adhocore** [comment]
excuse me, is something blocking this from being merged?

**Comment by GrahamCampbell** [comment]
@adhocore Yes, the weekend.

**Comment by nusphere** [comment]
some reason for this delay / block?

**Comment by WyriHaximus** [comment]
@nusphere It's Monday, nobody really likes Monday. Honestly no clue, but it's not really working hours yet in some parts of the world. Give them time, besides you can always use `8.0-rc` if you want to start building images with PHP 8 in it.

**Comment by Radiergummi** [comment]
Everyone is hyped for PHP 8, me too. But this is still an Open Source project, and none of us needs (or should, for that matter) migrate their production stack to a new major version on release day? Let people do their unpaid jobs as they deem fit, eat some cookies and relax 😉

**Comment by tianon** [comment]
> ... it's not really working hours yet in some parts of the world.

Just so happens that those "some parts of the world" are the same place where PHP's release day was on one of the most widely observed holidays of the entire year. :smile:

**Comment by FabianKoestring** [comment]
Is there a reason why `amd/64` for `8.0.0-fpm-buster` does not exists?

**Comment by panakour** [comment]
Also `8.0-fpm-alpine` on `linux/amd64`

**Comment by mruz** [comment]
I'm getting "no matching manifest":
> 8.0.0-fpm-alpine3.12: Pulling from library/php
Service 'php' failed to build : no matching manifest for linux/amd64 in the manifest list entries

**Comment by ondrejmirtes** [comment]
I'm also getting an error:

#3 [internal] load metadata for docker.io/library/php:8-cli-alpine
#3 ERROR: no match for platform in manifest sha256:f185d3f285621848688a84df864516087bcb163fa600afd49ac2e735cbb667b7: not found

**Comment by WyriHaximus** [comment]
@FabianKoestring @panakour @ondrejmirtes  It takes 8 to 12 hours to build the images for all architectures. As usual first come the ARM images before the AMD64 images start coming in, give it time and check again later today. Keep at eye on https://hub.docker.com/_/php/?tab=tags&page=1&ordering=last_updated&name=8.0.0- and make sure to expand the `OS/ARCH` column, it should show AMD64 in a few hours tops. (Note that the hours mentioned are per what I've been observing with previous builds of PHP Docker images.)

**Comment by rgeraads** [comment]
> It takes 8 to 12 hours to build the images for all architectures.

Out of curiosity: is that process visible somewhere?

**Comment by JoelLinn** [comment]
https://doi-janky.infosiftr.net/job/multiarch/job/amd64/job/php/ should pop up here. Log of latest build says it’s already pushed though🧐

**Comment by hollodotme** [comment]
Jep, it's there under https://hub.docker.com/r/amd64/php/tags?page=1&ordering=last_updated&name=8.0 but not yet under https://hub.docker.com/_/php/?tab=tags&page=1&ordering=last_updated&name=8.0

So `docker pull amd64/php:8.0-fpm-alpine` works just fine, but `docker pull php:8.0-fpm-alpine` doesn't yet.

**Comment by Carpenter0100** [comment]
amd64/php:8.0-fpm-alpine works fine. Thank you @hollodotme

**Comment by adhocore** [comment]
> As usual first come the ARM images before the AMD64 images start coming in

would it be nice if we could prioritize **linux/amd64** build before _arm_? (if possible)

**Comment by ondrejmirtes** [comment]
@adhocore That doesn't make sense, you just need to wait. Someone else might like that ARM comes first.

**Comment by WyriHaximus** [comment]
> > As usual first come the ARM images before the AMD64 images start coming in
> 
> would it be nice if we could prioritize **linux/amd64** build before _arm_? (if possible)

*shrug* Normally it doesn't matter, and it's only lagging an hour or two behind. We're now all here because this is the first GA image of PHP 8 and we're super excited, you wouldn't notice unless you target patch versions in your Dockerfiles.

**Comment by adhocore** [comment]
just looking at the restlessness of all of us commenting and reacting here yearning for amd, maybe it is more ubiquitous compared to arm (at least for where php is made use of, not speaking statistically though) ?
even docker hub build seems to require linux/amd arch.

**Comment by dungahk** [comment]
I'm not a maintainer of the repository or anything like that but I would recommend people to raise a separate issue if they think amd64 should be built/pushed before arm as it is unrelated to this PR goal/subject.

**Comment by adhocore** [comment]
> would it be nice if we could prioritize linux/amd64 build before arm? (if possible)

no. it would not. that's it. let's be done here.

**Comment by tianon** [comment]
We don't explicitly prioritize *either* architecture -- the build queues are fully independent of one another.

What actually happened here is that ARM managed to build and push faster than amd64 did, then we hit Docker's rate limits while trying to update the manifest lists for a while.

**Comment by tianon** [comment]
Also, all that has nothing to do with this repository; see https://github.com/docker-library/faq#an-images-source-changed-in-git-now-what for an outline of the full build/push process.

**Pull request merged**
